lib/spack/spack/package_base.py

```diff
@@ -952,6 +952,8 @@ def fullnames(cls):
             namespace = getattr(cls, "namespace", None)
             if namespace:
                 fullnames.append("%s.%s" % (namespace, cls.name))
+            # TODO/RepoSplit: Will the inability to inherit from other repos
```

Comment:
I think we can safely either delete this condition or check whether `cls is PackageBase` instead.

Reply:
🤔 Are you sure? I'd assumed the reason for disallowing builtin packages from inheriting from other repos is to avoid problems with having to clone multiple package repositories just to support what has been called the `builtin` package repository.

Reply:
This check isn't what protects from packages in builtin inheriting from somewhere else, it just relies on that to exit slightly earlier from an iteration over the mro. Since the classes above the repo don't have namespaces and a handful of `getattr` calls in a method that's only used in one other spot in the code base doesn't really have any performance impact, it might not be worth preserving.